### PR TITLE
Mangasail Rework

### LINKED ISSUE #5451: [MangaSail] Connector not working
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Is the website of the connector working properly / are you able to see the manga within your browser**
Yes

**Describe the bug**
Each time I press the refresh while MangaSail is the connector, the error "failed to update manga list" appears with "Cannot read property 'text of undefined" right below it
 
**To Reproduce**
1. MangaSail
2. Name of manga (if connector update is working) - Any
3. Name of chapter (if manga update is working) - Any
4. Copy any relevant error message that appears in a popup or (advanced) within the console (press F12)
image
image

**HakuNeko (please complete the following information):**
 - Version 1.6.7@554b0c

### PR DESCRIPTION
**Pull request by MikeZeDev**
* Fix URL
* New coding style
* Fix getting pages : remove bloat & add referer
* Fix getting chapters (now multipages)
* Fix login typo
* Fixes https://github.com/manga-download/hakuneko/issues/5451

### CODE REVIEW

File: src/web/mjs/connectors/MangaSail.mjs [hunk 1]
```diff
@@ -29,69 +29,61 @@ export default class MangaSail extends Connector {
...
-    _getMangaListFromPages( mangaPageLinks, index ) {
-        if( index === undefined ) {
-            index = 0;
+    async _getMangas() {
+        let mangaList = [];
+        const uri = new URL('/directory', this.url);
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'ul.pagination li.pager-last a');
+        const pageCount = parseInt(data[0].href.match(/([\d]+)$/)[1]);
+        for(let page = 0; page <= pageCount; page++) {
```
Comment (27136773) [inline src/web/mjs/connectors/MangaSail.mjs:38]: Page starts at `0` and not at `1` is this correct?

File: src/web/mjs/connectors/MangaSail.mjs [hunk 2]
```diff
@@ -29,69 +29,61 @@ export default class MangaSail extends Connector {
...
-                callback( error, undefined );
-            } );
+    async _getChapters(manga) {
+
+        let chapterList = [];
+        const uri = new URL (manga.id, this.url );
+        const request = new Request(uri, this.requestOptions);
+        const data = await this.fetchDOM(request, 'ul.pagination li.pager-last a');
+        const pageCount = data.length > 0 ? parseInt(data[0].href.match(/([\d]+)$/)[1]) : 0;
+        for(let page = 0; page <= pageCount; page++) {
```
Comment (27136773) [inline src/web/mjs/connectors/MangaSail.mjs:64]: Page starts at `0` and not at `1` is this correct?

**Review by 27136773 — APPROVED** [review]

File: src/web/mjs/connectors/MangaSail.mjs (hunk 1 above)
Comment (MikeZeDev) [reply src/web/mjs/connectors/MangaSail.mjs:38]: Correct. Link for page entitled "2" actually is https://www.mangasail.net/directory?page=1 , page "1" link is 0 etc...

File: src/web/mjs/connectors/MangaSail.mjs (hunk 2 above)
Comment (MikeZeDev) [reply src/web/mjs/connectors/MangaSail.mjs:64]: Same answer than for manga pages

**Pull request merged**